### PR TITLE
[SCYL][ROCM] Support for ROCm backend

### PR DESCRIPTION
**Pull request by AidanBeltonS**
Added support for targeting ROCm backend
Added new flag for -mcpu needed with sycl_triple for ROCm
Minor changes to some tests to for rocm
Updated readme to reflect the changed flags

Results from AMD GPU
Unsupported        :  99
Passed             : 183
Expectedly Failed  :   3
Failed             : 245
Unexpectedly Passed:   1

### CODE REVIEW

File: SYCL/Plugin/sycl-ls-gpu-rocm.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// REQUIRES: gpu, rocm
```
Comment (vladimirlaz) [inline SYCL/Plugin/sycl-ls-gpu-rocm.cpp]:
```suggestion
// REQUIRES: gpu, rocm, sycl-ls
```

File: SYCL/Plugin/sycl-ls-gpu-sycl-be.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// REQUIRES: gpu, cuda, opencl
+// REQUIRES: gpu, cuda, rocm, opencl
```
Comment (vladimirlaz) [inline SYCL/Plugin/sycl-ls-gpu-sycl-be.cpp]:
```suggestion
// REQUIRES: gpu, cuda, rocm, opencl, sycl-ls
```

File: SYCL/Plugin/sycl-ls-gpu-sycl-be.cpp [hunk 3]
```diff
@@ -15,6 +15,12 @@
 // CHECK-BUILTIN-GPU-CUDA: gpu_selector(){{.*}}GPU : CUDA
 // CHECK-CUSTOM-GPU-CUDA: custom_selector(gpu){{.*}}GPU : CUDA
 
+// RUN: env SYCL_DEVICE_FILTER=rocm sycl-ls --verbose >%t.rocm.out
+// RUN: FileCheck %s --check-prefixes=CHECK-BUILTIN-GPU-ROCM,CHECK-CUSTOM-GPU-ROCM --input-file %t.rocm.out
+
+// CHECK-BUILTIN-GPU-ROCM: gpu_selector(){{.*}}GPU : ROCM
```
Comment (vladimirlaz) [inline SYCL/Plugin/sycl-ls-gpu-sycl-be.cpp]: According to https://rocmdocs.amd.com/en/latest/ is should be ROCm not ROCM

File: SYCL/README.md [hunk 4]
```diff
@@ -81,18 +81,19 @@ list of configurations. Each configuration includes backend separated
...
 
 ```
--DCHECK_SYCL_ALL="opencl:cpu,host;level_zero:gpu,host;cuda:gpu"
+-DCHECK_SYCL_ALL="opencl:cpu,host;level_zero:gpu,host;cuda:gpu;rocm:gpu"
 ```
 
 ***SYCL_BE*** SYCL backend to be used for testing. Supported values are:
  - **opencl** - for OpenCL backend;
  - **cuda** - for CUDA backend;
+ - **ROCm** - for ROCm backend;
```
Comment (vladimirlaz) [inline SYCL/README.md]: rocm
```suggestion
 - **rocm** - for ROCm backend;
```
we use all-small here

File: SYCL/README.md [hunk 5]
```diff
@@ -105,6 +106,15 @@ specified by this variable.
 
 ***LEVEL_ZERO_LIBS_DIR*** path to Level Zero libraries.
 
+***ROCM_PLATFORM*** platform selection for ROCm targeted devices.
+Defaults to AMD if no value is given. Supported values are:
+ - **AMD**    - for ROCm to target AMD GPUs
+ - **NVIDIA** - for ROCm to target NVIDIA GPUs
+ 
+ ***MCPU*** Please not, flag must be set for when using ROCm triple.
```
Comment (vladimirlaz) [inline SYCL/README.md]: It look slike misprinted sentence

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: SYCL/Plugin/sycl-ls-gpu-rocm.cpp (hunk 1 above)
Comment (AidanBeltonS) [reply SYCL/Plugin/sycl-ls-gpu-rocm.cpp]: All changes have been implemented.
clang format has been run

**Review by vladimirlaz — DISMISSED** [review]
LGTM

**Review by vladimirlaz — APPROVED** [review]

**Comment by AidanBeltonS** [comment]
Is it possible to find out what aspect has caused the pre-ci-windows test to fail?

**Comment by vladimirlaz** [comment]
The fallures on Windows are unrelated to the changes.

**Comment by vladimirlaz** [comment]
> Is it possible to find out what aspect has caused the pre-ci-windows test to fail?

The windows failures have been addressed in https://github.com/intel/llvm-test-suite/pull/364.

**Pull request merged**